src/source/source_cache.js

```diff
@@ -652,11 +678,29 @@ class SourceCache extends Evented {
         return tileResults;
     }
 
-    redoPlacement() {
+    placeLayer(collisionIndex: CollisionIndex, showCollisionBoxes: boolean, layer: any, posMatrices: any, shouldPausePlacement: any) {
+        while (this._placementIndex < this._placementIDs.length) {
+            const tile = this.getTileByID(this._placementIDs[this._placementIndex]);
+            if (!posMatrices[this._placementIndex]) {
+                posMatrices[this._placementIndex] = collisionIndex.transform.calculatePosMatrix(tile.coord, tile.sourceMaxZoom);
+            }
```

Comment:
Would it make sense to move this caching to the inside of `transform.calculatePosMatrix(...)`? not sure

Reply:
I like that idea, I'll give it a try.

Reply:
Done.